### ISSUE TITLE
Structure of the crate

### ISSUE THREAD

**Issue by Darksonn** [ploteria/ploteria#1]
There seems to be a lot of types all over the main module, as well as a bunch of modules with one or two types. As I far as I can see the content in the crate categorize into:

 * Types used as the first argument in configure: Axis, Grid, Key.
 * Values used for settings: BoxWidth, Font, FontSize, Label, LineWidth, Opacity, Output, PointSize, ScaleFactor, Size, TicLabels, Title, Axes, Color, LineType, PointType, Range, Scale, Terminal, key::Boxed, key::Justification, key::Order, key::Position, key::Stacked.
 * Types used inside types used for settings: key::Horizontal and key::Vertical (both used in key::Position).
 * A bunch of types all named properties: axis::Properties, candlestick::Properties, curve::Properties, errorbar::Properties, filledcurve::Properties, grid::Properties, key::Properties.
 * Things that can be plotted: candlestick::Candlesticks, curve::Curve, errorbar::Errorbar, filledcurve::FilledCurve.
 * Some functions that look like constructors: proxy::Font, proxy::Label, proxy::Output, proxy::Title.
 * Four traits: traits::Configure, traits::Data, traits::Plot, traits::Set.
 * Other stuff: Figure, VersionError.

I believe the current organization is pretty messy, and we should probably clean it up.

PS: I recommend viewing the documentation from a locally compiled `cargo doc` while also running `cargo test`, as otherwise images are not displayed in the doc.

**Comment by Darksonn** [comment]
In my opinion we should consider removing the `Set` trait and replacing the implementations with concrete methods. To see why I propose this, consider the following snippet from an example of the doc:

```rust
lp.set(Color::DarkViolet)
  .set(Label("sin(x)"))
  .set(LineType::Dash)
  .set(PointSize(1.5))
  .set(PointType::Circle)
```

While this code looks nice and readable, I want to propose changing the library to this:

```rust
lp.set_color(Color::DarkViolet)
  .set_label("sin(x)")
  .set_line_type(LineType::Dash)
  .set_point_size(1.5)
  .set_point_type(PointType::Circle)
```

The readability of the code is about the same, but it vastly improves the readability of the documentation for these setters, which currently looks like this:

![Screenshot from Figure](https://user-images.githubusercontent.com/928074/51704384-395a4800-2019-11e9-9c76-e06cb6f44b6f.png)

As you can see all setters on the `Figure` type are hidden in the traits section, which is both quite unusual and also has the problem that you need to scroll pretty far down to find them, and you also have to manually open the `impl` section of every setter to see its documentation:

![Screenshot from Figure](https://user-images.githubusercontent.com/928074/51704502-85a58800-2019-11e9-87fa-8bd4d0eac807.png)

Finally we get to remove the `proxy` module as well as a bunch of the types in the "Values used for settings" category above.

**Comment by Darksonn** [comment]
The same documentation problem occurs with `configure`, but I'm not sure what the best alternatives are. The code also seems to be less clear than it should be, for example look at this snippet from the examples:

```rust
Figure::new().
    // ... other stuff
    configure(Axis::RightY, |a| a.
        configure(Grid::Major, |g| g.
            show()).
        set(Label("Phase shift (°)"))).
```

It doesn't seem to too clear what's going on here, and while that might be improved by moving around some newlines and indents, I'm especially not convinced the nested calls to `configure` is a good solution.

Maybe something like this would be reasonable, although it's not as friendly to the chained builder pattern.

```rust
figure.configure_axis(Axis::RightY)
    .set_label("Phase shift (°)")
    .show_grid(Grid::Major);
```

**Comment by mfep** [comment]
I'd suggest closing this issue, as the main points were addressed at #6 and #8.
